sources/packages/backend/apps/queue-consumers/src/services/cas-invoice-batch/cas-invoice-batch.service.ts

```diff
@@ -0,0 +1,250 @@
+        "CAS_INVOICE_BATCH",
+        entityManager,
+        async (nextSequenceNumber: number) => {
+          newBatchUniqueSequence = nextSequenceNumber;
+        },
+      );
+      processSummary.info("Checking for pending receipts.");
+      const pendingReceipts = await this.getPendingReceiptsForInvoiceBatch(
+        entityManager,
+      );
+      if (!pendingReceipts.length) {
+        processSummary.info("No pending receipts found.");
+        // Cancels the transaction to rollback the batch number.
+        await entityManager.queryRunner.rollbackTransaction();
+        return null;
+      }
+      processSummary.info(`Found ${pendingReceipts.length} pending receipts.`);
+      const now = new Date();
+      // Create the new invoice batch.
+      const invoiceBatch = new CASInvoiceBatch();
+      invoiceBatch.batchName = `SIMS-BATCH-${newBatchUniqueSequence}`;
+      invoiceBatch.batchDate = now;
+      invoiceBatch.approvalStatus = CASInvoiceBatchApprovalStatus.Pending;
+      invoiceBatch.approvalStatusUpdatedOn = now;
+      invoiceBatch.approvalStatusUpdatedBy = this.systemUsersService.systemUser;
+      invoiceBatch.creator = this.systemUsersService.systemUser;
+      // Save the invoice batch to allow the ID to be known and associate with every new
+      // invoice created. It would allow the invoices to be saved in chunks.
+      await entityManager.getRepository(CASInvoiceBatch).save(invoiceBatch);
+      // Accumulates the invoices to be returned later.
+      const casInvoices: CASInvoice[] = [];
+      // Get active distribution accounts to be associated with each invoice detail.
+      const distributionAccounts = await this.getActiveDistributionAccounts(
+        entityManager,
+      );
+      await this.sequenceControlService.consumeNextSequenceWithExistingEntityManager(
+        "CAS_INVOICE",
+        entityManager,
+        async (nextSequenceNumber: number) => {
+          let newInvoiceSequence = nextSequenceNumber;
+          const invoicesSummary = new ProcessSummary();
+          processSummary.children(invoicesSummary);
+          // Start processing the invoices for each pending receipt.
+          for (const receipt of pendingReceipts) {
+            invoicesSummary.info(
+              `Creating invoice for receipt ID ${receipt.id}.`,
+            );
+            const student =
+              receipt.disbursementSchedule.studentAssessment.application
+                .student;
+            const offeringIntensity =
+              receipt.disbursementSchedule.studentAssessment.offering
+                .offeringIntensity;
+            // Create a new invoice for the receipt.
+            const newInvoice = new CASInvoice();
+            newInvoice.casInvoiceBatch = invoiceBatch;
+            newInvoice.disbursementReceipt = receipt;
+            newInvoice.casSupplier = student.casSupplier;
+            newInvoice.invoiceNumber = `SIMS-INVOICE-${newInvoiceSequence++}-${
```

Comment:
Let me know if I am missing something here. When we keep incrementing the number obtained from sequence, will it not conflict on next consumption of the sequence?

Reply:
The sequence control service method was updated to allow it.

Reply:
I saw that later. Thanks.